stripe/src/main/java/com/stripe/android/view/MaskedCardView.java

```diff
@@ -0,0 +1,270 @@
+    String getCardBrand() {
+        return mCardBrand;
+    }
+
+    @VisibleForTesting
+    String getLast4() {
+        return mLast4;
+    }
+
+    void init() {
+        inflate(getContext(), R.layout.masked_card_view, this);
+        setOrientation(HORIZONTAL);
+        setMinimumWidth(getResources().getDimensionPixelSize(R.dimen.card_widget_min_width));
+        int paddingPixels = getContext()
+                .getResources().getDimensionPixelSize(R.dimen.masked_card_vertical_padding);
+        setPadding(0, paddingPixels, 0, paddingPixels);
+
+        mCardIconImageView = findViewById(R.id.masked_icon_view);
+        mCardInformationTextView = findViewById(R.id.masked_card_info_view);
+        mCheckMarkImageView = findViewById(R.id.masked_check_icon);
+
+        mSelectedColorInt = getThemeAccentColor(getContext()).data;
+        mUnselectedColorInt = getThemeColorControlNormal(getContext()).data;
+        useDefaultColorsIfThemeColorsAreInvisible();
+
+        initializeCheckMark();
+        updateCheckMark();
+    }
+
+    private void initializeCheckMark() {
+        updateDrawable(R.drawable.ic_checkmark, mCheckMarkImageView, true);
+    }
+
+    private void updateBrandIcon() {
+        @DrawableRes int iconResourceId = TEMPLATE_RESOURCE_MAP.get(mCardBrand);
+        updateDrawable(iconResourceId, mCardIconImageView, false);
+    }
+
+    private void updateTextColor() {
+        @ColorInt int textColor = mIsHighlighted ? mSelectedColorInt : mUnselectedColorInt;
+        mCardInformationTextView.setTextColor(textColor);
+    }
+
+    @SuppressWarnings("deprecation")
+    private void updateDrawable(
+            @DrawableRes int resourceId,
+            @NonNull ImageView imageView,
+            boolean isCheckMark) {
+        Drawable icon;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            icon = getResources().getDrawable(resourceId, imageView.getContext().getTheme());
+        } else {
+            // This method still triggers the "deprecation" warning, despite the other
+            // one not being allowed for SDK < 21
+            icon = getResources().getDrawable(resourceId);
+        }
+
+        @ColorInt int tintColor = mIsHighlighted || isCheckMark ? mSelectedColorInt : mUnselectedColorInt;
+        Drawable compatIcon = DrawableCompat.wrap(icon);
+        DrawableCompat.setTint(compatIcon.mutate(), tintColor);
+        imageView.setImageDrawable(compatIcon);
+    }
+
+    private void updateCardInformation() {
+        String brandText = Card.AMERICAN_EXPRESS.equals(mCardBrand)
+                ? getResources().getString(R.string.amex_short)
+                : mCardBrand;
+        String normalText = getResources().getString(R.string.ending_in);
+        int brandLength = brandText.length();
+        int middleLength = normalText.length();
+        int last4length = mLast4.length();
+        SpannableString str = new SpannableString(brandText + normalText + mLast4);
```

Comment:
Just wondering generally, should we switch to using a language-agnostic representation instead of saying "Ending In"? I'm thinking it should be something like:

`[MasterCard] MasterCard ****8210`.

I guess I don't completely understand why I feel like that is better. One reason might be because it feels quicker to visually parse symbols/icons than reading text.